### DISCUSSION TITLE
Remove tags from a virtual machine

### DISCUSSION THREAD

**Discussion by BoHu1977** [netbox-community/netbox#10893] [Q&A]
How can I remove 1 or more tag(s) from a virtual machine? I use the API virtualization \ virtual machine with code for the action. 

Part of the code:
$TAGS["tags"] = $null
$UpdateNetBoxVM ConvertTo-JSON $TAGS
$URI = $URIBase + "/virtualization/virtual-machines" + "/<ID>/"
$Respons = Invoke-RESTmethod -Method PATCH -Headers $Headers -ContentType "application/json" -Body $UpdateNetBoxVM -URI $URI

**Comment by candlerb** [comment]
Please show your existing code and what you've tried.  Presuming you are talking about the REST API, are you using `pynetbox` or `requests` or `curl` or some other client library?

**Reply by BoHu1977** [reply]
Added code

**Reply by candlerb** [reply]
I think that code has been mangled by markdown rendering.  Please can you edit the post and put three backticks (` ``` `) on a line of their own, before and after the code.

I'm not sure what language it is: PowerShell perhaps?

I think you need to set the tags to the empty list - `[]` - instead of Null.

If you can show the JSON contents of the request body being sent, which I think is whatever is stored in `$UpdateNetBoxVM`, that would make it clearer.

**Reply by BoHu1977** [reply]
The language is indeed PowerShell.
I Tried the [], but it fails. The error code ``` Invoke-RESTMethod: {"details:""Not found"}``` is show
The content of ```UpdateNetBoxVM``` is a System.Collections.DictonaryEntry with the content of 
```{```
```"tags": null```
```} ```

**Reply by candlerb** [reply]
And can you show the content of UpdateNetBoxM when you set tags to empty list?

**Reply by candlerb** [reply]
Also, *please* can you edit your first post to add the three-backticks lines for quoting your code block.

Right now I see `$URI = $URIBase + "/virtualization/virtual-machines" + "//"` and that can't be right.

**Reply by candlerb** [reply]
Tested with curl it works fine (Netbox 3.3.7), so it must be that your PowerShell fu is the problem.

```
# curl -gsS -H "Authorization: Token $API_TOKEN" "https://$NETBOX/api/virtualization/virtual-machines/8/" | jq '.tags'
[]

# curl -d '{"tags":[9]}' -X PATCH -gsS -H "Content-Type: application/json" -H "Authorization: Token $API_TOKEN" "https://$NETBOX/api/virtualization/virtual-machines/8/"
... snip response ...

# curl -gsS -H "Authorization: Token $API_TOKEN" "https://$NETBOX/api/virtualization/virtual-machines/8/" | jq '.tags'
[
  {
    "id": 9,
    "url": "https://XXXX/api/extras/tags/9/",
    "display": "gateway",
    "name": "gateway",
    "slug": "gateway",
    "color": "009688"
  }
]

# curl -d '{"tags":[]}' -X PATCH -gsS -H "Content-Type: application/json" -H "Authorization: Token $API_TOKEN" "https://$NETBOX/api/virtualization/virtual-machines/8/"
... snip response ...

# curl -gsS -H "Authorization: Token $API_TOKEN" "https://$NETBOX/api/virtualization/virtual-machines/8/" | jq '.tags'
[]
```
